README.rst

```diff
@@ -158,5 +176,217 @@ using the ``gridded_perms_of_length`` method.
+       >>> spec = tilescope.auto_search()
+       >>> print(spec)
+       A combinatorial specification with 5 rules.
+
+       0 -> (1, 2)
+       Explanation: insert 0 in cell (0, 0)
+       +-+            +-+     +-+
+       |1|         =  | |  +  |1|
+       +-+            +-+     +-+
+       1: Av(120)             1: Av+(120)
+                            Requirement 0:
+                            0: (0, 0)
+
+       1 -> ()
+       Explanation: is atom
+       +-+
+       | |
+       +-+
+
+
+       2 = 3
+       Explanation: placing the topmost point in cell (0, 0), then row and column separation
+       +-+                +-+-+-+                    +-+-+-+
+       |1|             =  | |●| |                 =  | |●| |
+       +-+                +-+-+-+                    +-+-+-+
+       1: Av+(120)        |1| |1|                    | | |1|
+       Requirement 0:     +-+-+-+                    +-+-+-+
+       0: (0, 0)          1: Av(120)                 |1| | |
+                            ●: point                   +-+-+-+
+                            Crossing obstructions:     1: Av(120)
+                            10: (0, 0), (2, 0)         ●: point
+                            Requirement 0:             Requirement 0:
+                            0: (1, 1)                  0: (1, 2)
+
+       3 -> (0, 4, 0)
+       Explanation: factor with partition {(0, 0)} / {(1, 2)} / {(2, 1)}
+       +-+-+-+            +-+            +-+                +-+
+       | |●| |         =  |1|         x  |●|             x  |1|
+       +-+-+-+            +-+            +-+                +-+
+       | | |1|            1: Av(120)     ●: point           1: Av(120)
+       +-+-+-+                           Requirement 0:
+       |1| | |                           0: (0, 0)
+       +-+-+-+
+       1: Av(120)
+       ●: point
+       Requirement 0:
+       0: (1, 2)
+
+       4 -> ()
+       Explanation: is atom
+       +-+
+       |●|
+       +-+
+       ●: point
+       Requirement 0:
+       0: (0, 0)
+
+Now that we have a specification we can do a number of things. For example,
+counting how many permutations there are in the class. This can be done using
+the ``count_objects_of_size`` method on the CombinatorialSpecification.
+
+.. code:: python
+
+       >>> [spec.count_objects_of_size(i) for i in range(10)]
+       [1, 1, 2, 5, 14, 42, 132, 429, 1430, 4862]
+
+Of course we see the Catalan numbers! We can also sample uniformly using the
+``random_sample_object_of_size`` method. This will return a ``GriddedPerm``. If
+you want the underlying ``Perm``, this can be accessed with ``patt`` attribute.
+I have done this here, and then used the ``permuta.Perm.ascii_plot`` method for
+us to visualise it.
+
+.. code:: python
+
+       >>> gp = spec.random_sample_object_of_size(10)
+       >>> perm = gp.patt
+       >>> print(perm)
+       5042136987
+       >>> print(perm.ascii_plot())
+       | | | | | | | | | |
```

Comment:
The line of the grid are not aligned properly. Is this a bug on permuta?

Reply:
No, just a copy-paste error